pkg/logger/zap.go

```diff
@@ -0,0 +1,82 @@
+package logger
+
+import (
+	"sync"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// keep the logger in a singleton mode, allowing for
+// dynamic changes to the log level at runtime.
+var (
+	zapLogLevel zap.AtomicLevel
+	zapLogger   *zap.SugaredLogger
+	once        sync.Once
+)
+
+// InitLogger initializes the logger with the given environment.
+// Must be called before using the logger.
+func InitLogger(env string) {
```

Comment:
Could we combine InitLogger and GetLogger into new GetLogger:
```suggestion
func GetLogger() *zap.SugaredLogger {
if zapLogger == nil {
		zapLogLevel = zap.NewAtomicLevel()
		zapConfig := zap.NewDevelopmentConfig()
		zapConfig.DisableStacktrace = true
		zapConfig.Encoding = "console"
		if environments.GetEnvironmentStrFromEnv() == "development" {
			zapLogLevel.SetLevel(zapcore.DebugLevel)
		} else {
			zapLogLevel.SetLevel(zapcore.InfoLevel)
		}
		zapLogLevel.SetLevel(zapcore.InfoLevel)
		zapConfig.Level = zapLogLevel
		zlog, err := zapConfig.Build()
		if err != nil {
			panic(err)
		}

		zapLogger = zlog.Sugar()
	}

	return zapLogger
}
```

Reply:
We can't merge `InitLogger` and `zapLogger` because `InitLogger` is called by the framework before the environment is initialized. `GetLogger()` is used in every package that needs to log messages. The `zapLogger == nil` check is necessary because, in unit tests, we don't use the framework, so it doesn't call `InitLogger`, in that case we initialize the logger in `GetLogger()`.

Reply:
merged `InitLogger()` and `GetLogger()` as we discussed offline.